.github/workflows/ci.yml

```diff
@@ -69,20 +70,29 @@ jobs:
       # XXX this should be instant because all the stages were already built in the steps above
       # but it's re-building the last stage. This is deemed good enough for now.
       # see https://github.com/moby/moby/issues/34715
-      - name: Build and push complete image
+      - name: Build and push native-only image
         uses: docker/build-push-action@v5
+        if: "github.event_name == 'pull_request' || github.event_name == 'merge_group'"
         with:
           context: .
           # push to registry on every repo push. A PR #2 will push with tag `pr-2` and `main` will have tag `main`.
           # See https://github.com/docker/metadata-action?tab=readme-ov-file#basic.
           push: true
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
-      - name: notify on failure
-        if: failure() && github.event_name != 'pull_request'
-        uses: ./.github/actions/notify-status
+
+      - name: Set up QEMU for cross-platform builds
```

Comment:
what would it take to use the M1 runner? https://github.blog/2023-10-02-introducing-the-new-apple-silicon-powered-m1-macos-larger-runner-for-github-actions/

Reply:
That runner is a bit of a fortune. Though not sure how much a large runner in emulation mode would cost.

Reply:
We only need to build with it once per passed proposal. The announcement says $0.12/minute so $10 buys 83 minutes. That's probably enough.

Of course if QEMU works fine and it's only for `main`, then waiting a long time is not so bad.

Reply:
I see that the non-`main` and non-`workflow_dispatch` case doesn't take long to build.  @turadg or @mhofman, do you know why I don't see any way to trigger the `workflow_dispatch` from Github UI?  I'm not an admin on this repo, FWIW.

If somebody with privs could issue a `.github/workflows/ci.yml` `workflow_dispatch` for this `mfig-multiplatform-docker-main` branch, that would be great as a comparison with this PR's single-platform build.

Reply:
> I'm not an admin on this repo, FWIW.

You and @mhofman are now.

I don't know about the rest

Reply:
GH only adds the trigger UI once the workflow has `workflow_dispatch` in master, aka it should show up once this PR merges.